src/apiProxy/api-proxy.js

```diff
@@ -27,20 +37,28 @@ import { duotoneRef } from '../util/duotone';
  * @param response {String} the raw response body text from an API call
  * @return responseObj the JSON-parsed text, possibly with error info
  */
-export const parseApiResponse = response => {
-	let responseObj;
-	let error;
+export const parseApiResponse = ([response, body]) => {
+	let value;
+	const flags = parseResponseFlags(response);
 
-	try {
-		responseObj = JSON.parse(response);
-	} catch(e) {
-		error = `API response was not JSON: "${response}"`;
+	if (response.statusCode > 299) {
+		return {
+			value: { error: response.statusMessage },
+			flags,
+		};
 	}
-	if (responseObj && responseObj.problem) {
-		error = `API problem: ${responseObj.problem}: ${responseObj.details}`;
+	try {
+		value = JSON.parse(body);
+		if (value && value.problem) {
+			value = { error: `API problem: ${value.problem}: ${value.details}` };
+		}
+		return { value, flags };
+	} catch(err) {
+		return {
+			value: { error: err.message },
```

Comment:
Another nitpick, might be a bit DRY-er to do `return formatErrorResponse(...)` for the three places that we're returning errors in this function. i.e., in case we ever want to update what the error response object looks like